Tests/Client/RequestRecorderURLProtocol.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  MockURLProtocol.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+/// This URLProtocol subclass allows to intercept the network communication
+/// and provides the latest network request made.
+class RequestRecorderURLProtocol: URLProtocol {
+
+    /// Returns the latest network request this URLProtocol recorded.
+    ///
+    /// If no request has been made since the last time this function was invoked, it synchronously
+    /// waits for the next request to be made.
+    ///
+    /// - Parameter timeout: Specifies the time the function waits for a new request
+    /// to be made.
+    static func getLatestRequestSynchronously(timeout: TimeInterval) -> URLRequest? {
+        defer {
+            // Delete the used request
+            latestRequest = nil
+        }
+
+        guard latestRequest == nil else { return latestRequest }
+
+        latestRequestExpectation = .init(description: "Wait for incoming request.")
+        _ = XCTWaiter.wait(for: [latestRequestExpectation!], timeout: 1)
```

Comment:
`timeout` parameters is not used and hardcoded as `1`, I suggest providing default `timeout` as 1

Reply:
Still doesn't use `timeout`.

Reply:
@buh That isn't that surprising since there wasn't any new commit in the branch in the last 23 hours, hm? :trollface:

Reply:
I saw the `timeout` parameter and thought you added, but forgot to use it.

Reply:
fixed